[Then Jan Kai] iP

jan-kai1 opened the pull request:
# Chatterbox :smiley:
> Best chatbot since duke

It is 
1.  Fast
2. Text based 

Adds
- Commands `todo deadline list`
- Storage
- Date time

*More* to come!
- [ ] GUI

Find out more [here](www.google.com)
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jan-kai1 committed:
test level 0

jan-kai1 committed:
tag test

jan-kai1 committed:
testing

jan-kai1 committed:
echo done

jan-kai1 committed:
list done

jan-kai1 committed:
mark and unmark

jan-kai1 committed:
event, deadline, todo

jan-kai1 committed:
automated test?

jan-kai1 committed:
added invalid command, missing input and missing parameter errors

jan-kai1 committed:
added delete

jan-kai1 committed:
testing tags

jan-kai1 committed:
readd tag

jan-kai1 committed:
adds enum

jan-kai1 committed:
add enums

jan-kai1 committed:
adds save, creates data directory where program is run to track history

jan-kai1 committed:
Merge branch 'save'

jan-kai1 committed:
adds date and time parsing for deadline and event

jan-kai1 committed:
Merge branch 'dateTime'

jan-kai1 committed:
adds Ui parser and Tasklist

jan-kai1 committed:
fixes text formatting

jan-kai1 committed:
adds storage and tasklist objects, finishes implementation of basic oop objects

jan-kai1 committed:
adds missing load feature, moves running of chatbox and implements initialization of Chatterbox object to run

jan-kai1 committed:
resolves merge conflict between master and oop

jan-kai1 committed:
moves Task and subclasses, Parser, Storage, UI into packages as well as placed Chatterbox in a package

jan-kai1 committed:
Merge branch 'packages'

jan-kai1 committed:
fixes missing package statement on main chatterbox.java

jan-kai1 committed:
updates .gitignore to ignore .class files

jan-kai1 committed:
removes previously tracked .class files before gitignore changes

jan-kai1 committed:
Merge branch 'add-gradle-support'

jan-kai1 committed:
Adds gradle support for build and running of Chatterbox.

jan-kai1 committed:
Adds basic jar file

jan-kai1 committed:
adds basic tests for Parser class

jan-kai1 committed:
adds more parser testcases

jan-kai1 committed:
adds test for Storage class

jan-kai1 committed:
Merge branch 'unittests'

jan-kai1 committed:
merged unit test and master

jan-kai1 committed:
adds javadoc comments for Task and subclasses

jan-kai1 committed:
changes formatting

jan-kai1 committed:
removes whitespace

jan-kai1 committed:
fixes merging with Level-7

jan-kai1 committed:
fixes typo in test

jan-kai1 committed:
changes branch name for tracking

jan-kai1 committed:
changes some comments

jan-kai1 committed:
resolves merge conflict with Level-8

jan-kai1 committed:
adds javadoc documentation for most used classes and methods

jan-kai1 committed:
fixes Chatterbox.java coding standard violations

jan-kai1 committed:
changes storage class HISTFILE to be final

jan-kai1 committed:
fixes violations in main Chatterbox class

jan-kai1 committed:
adds find command

jan-kai1 committed:
Merge branch 'branch-A-JavaDoc'

jan-kai1 committed:
resolve merge conflicts with CodingStandard branch

jan-kai1 committed:
cleans up code in Parser

jan-kai1 committed:
resolves merge conflict with Level-9

jan-kai1 committed:
Adds config file for checkstyle

jan-kai1 committed:
Resolves checkstyle errors in main

jan-kai1 committed:
Resolve checkstyle errors in test

jan-kai1 committed:
Merge branch 'A-Checkstyle'

jan-kai1 committed:
Starts implementation of gui

jan-kai1 committed:
Starts integration of chatterbox with gui

jan-kai1 committed:
Adds basic implementation of gui

jan-kai1 committed:
fixes storage interaction with gui

KrashKart commented on src/main/java/chatterbox/Chatterbox.java:
```diff
@@ -0,0 +1,320 @@
+import tasks.Task;
+import tasks.Todo;
+import ui.UI;
+
```
i like the ordering of your imports!

KrashKart commented on src/main/java/chatterbox/Chatterbox.java:
```diff
@@ -0,0 +1,320 @@
+
+    /**
+     * initiates Chatterbox with a prior history filepath
+     * @param filepath contains the history of tasks
```
hmm, perhaps you could add punctuation behind every line in your Javadocs?

KrashKart commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,245 @@
+     * @return the int in the input string
+     */
+    public int extractNum(String input) {
+        int length = input.length();
```
Nice naming of your functions and use of camelCase!

KrashKart commented on src/main/java/tasks/Event.java:
```diff
@@ -0,0 +1,75 @@
+            return super.getDescription() + String.format(" ( from %s to %s )",
+                    this.startDateObj.format(parser.Parser.getPrintDateFormatter()),
+                    this.endDateObj.format(parser.Parser.getPrintDateFormatter()));
+        }
```
I like that you continued the function call in a new line to avoid an excessively wide line of code!

KrashKart reviewed (commented):
LGTM! Perhaps some naming nits from me :+1:

hpcman commented on src/main/java/gui/ChatterboxResponses.java:
```diff
@@ -0,0 +1,128 @@
+ */
+public class ChatterboxResponses {
+    private static final String LINE_SEPARATOR = "____________________________________________________________";
+    private static final String BOTNAME = "Chatterbox";
+
+    private static final DateTimeFormatter PRINTDATEFORMATTER = DateTimeFormatter.ofPattern("MMM dd yyyy, HH:mm");
```
Would be better if <code>BOTNAME</code> changed to <code>BOT_NAME</code> and <code>PRINTDATEFORMATTER</code> to <code>PRINT_DATE_FORMATTER</code> for consistency

hpcman commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,245 @@
+    }
+
+
+    private static final DateTimeFormatter DASHFORMATTER = DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm");
+    private static final DateTimeFormatter SLASHFORMATTER = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+
+    private static final DateTimeFormatter DASHONLYDATE = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+    private static final DateTimeFormatter SLASHONLYDATE = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+    private static final DateTimeFormatter PRINTDATEFORMATTER = DateTimeFormatter.ofPattern("MMM dd yyyy, HH:mm");
+    private static final DateTimeFormatter[] DATE_TIME_FORMATTERS =
+            new DateTimeFormatter[] {DASHFORMATTER, SLASHFORMATTER, PRINTDATEFORMATTER};
+    private static final DateTimeFormatter[] DATE_ONLY_FORMATTERS =
+            new DateTimeFormatter[] {DASHONLYDATE, SLASHONLYDATE};
```
A bit inconsistent, it is better to use <code>DASH_ONLY_DATE</code> instead of <code>DASHONLYCODE</code> for consistency. This also applies for another static variable

hpcman commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,182 @@
+    }
+
+    /**
+     * Loads the
```
I think this statement is incomplete

hpcman reviewed (commented):
Overall LGTM!

jan-kai1 committed:
adds some asserts

jan-kai1 committed:
Adds asserts for major classes

This commit intoroduces asserts to key functions to enforce valid inputs
and outputs

- Asserts added to Parser and Storage Classes

These asserts improve code reliability by ensuring assumptions about
data are explicitly enforced, improving debugging.

jan-kai1 committed:
Merge pull request #1 from jan-kai1/A-Assertions

Merges Assertations branch

jan-kai1 committed:
adds additional assertations

jan-kai1 committed:
Merge pull request #2 from jan-kai1/branch-A-Assertions

Fixes branch naming

jan-kai1 committed:
This commit improves code quality by refactoring
and enhancing readability

- Abstracted TaskList to separate Class
- Shortened method lengths by introducing inner methods

These changes aim to make the codebase more maintainable, readable, and easier to understand, facilitating future development and debugging.

jan-kai1 committed:
Merge pull request #3 from jan-kai1/branch-A-CodeQuality

Merges the code quality branch

jan-kai1 committed:
adds basic tagging functionality

jan-kai1 committed:
Merge branch 'branch-C-Tagging'

jan-kai1 committed:
Implement tagging functionality

Added an ArrayList<String> for tags to the Task class to hold a list of tag strings.
Enabled tagging of tasks to categorize and manage them more effectively.
Updated relevant methods to support adding, removing, and retrieving tags from tasks.

jan-kai1 committed:
fix error with previous tag string

jan-kai1 committed:
implements basic adding of tag class

jan-kai1 committed:
fixes issue with event parsing

jan-kai1 committed:
refactors load method in Storage class, does not resolve loading errors for tags

jan-kai1 committed:
Add basic implementation of Tag and TagList object

- Implements Tag and TagList Object
- Integrate Tag and TagList with Storage and Parser
- Added additional test cases for parsing events and deadlines
- Fixes input bugs with event and deadline

Adding the Tag and TagList objects while increasing program complexity
allows for greater scalability with searching for tags.

jan-kai1 committed:
Merge branch 'tags'

jan-kai1 committed:
add command classes up till Find

jan-kai1 committed:
Improve Code Quality by adding Command class

Command class added to extract execution of different commands. This
improves code quality and readability, while adding abstraction for
adding new commands.

jan-kai1 committed:
Merge branch 'commands'

jan-kai1 committed:
varies response messages

jan-kai1 committed:
change basic responses to adde emotions

jan-kai1 committed:
Merge branch 'emotions'

jan-kai1 committed:
add support for case insensitive command

jan-kai1 committed:
Merge branch 'casing'

jan-kai1 committed:
add guards against missing parameter in tag commands

jan-kai1 committed:
rename various tag parsing methods for clarity

jan-kai1 committed:
Add guards against invalid input for parsing

- Checks for input string for task
- Checks for input index for task

jan-kai1 committed:
Add guard clauses against invalid indices

Add checks in the following Commands
- Delete
- Mark
- RemoveTag
- Tag
- Unmark

Adding indices checking guards against invalid user inputs which could
result in unexpected behaviour

jan-kai1 committed:
change error message to fit personality

jan-kai1 committed:
Merge branch 'tags'

jan-kai1 committed:
Add Additional Test for Parser and Storage

- Parser Class testing for valid Command Class return
- Storage Date Time loading

The additional unit test ensure no unexpected behaviour occurs due to
refactoring or changing of code

jan-kai1 committed:
fix issue with storage string formatting

jan-kai1 committed:
Change Storage load handling of invalid input

Storage now skips lines in the input file if the line cannot be parsed
into a task. Also fix an issue regarding untagged deadlines loading

This commit enhances the robustness of the storage system by improving error handling for invalid input
and addressing a bug related to untagged deadlines.
The system now skips lines that cannot be parsed into tasks, and untagged deadlines are correctly processed.

jan-kai1 committed:
Merge branch 'testing'

jan-kai1 committed:
Implements personality of Chatterbox

- Changes all guiResponses with responses that align with Chatterbox
  personality

Chatterbox personality is calm and introspective, with a touch of
melancholy.

jan-kai1 committed:
clean up unused code

jan-kai1 committed:
Merge branch 'A-Personality'

jan-kai1 committed:
clean up empty space

jan-kai1 committed:
add userguide

jan-kai1 committed:
add basic readme

jan-kai1 committed:
add basic readme structure

jan-kai1 committed:
fix ui for screenshot

jan-kai1 committed:
fix readme location

jan-kai1 committed:
fix issue with readme

jan-kai1 committed:
add additional DateTimeFormatter for date parsing

jan-kai1 committed:
add additional ui tweaks

jan-kai1 committed:
resize windows

jan-kai1 committed:
Revert "add additional ui tweaks"

This reverts commit 3452f20e25f81a2c5bd8d6484ab73ab043056cc9.

jan-kai1 committed:
fix window sizing

jan-kai1 committed:
fix expandable window formatting

jan-kai1 committed:
add window icon

jan-kai1 committed:
cleans up style errors

jan-kai1 committed:
Add additional date time parsing formatter

jan-kai1 committed:
add missing responses to task number

jan-kai1 committed:
Provide ui.png and move README

jan-kai1 committed:
add a tag for final publish

jan-kai1 committed:
readme fixes

jan-kai1 committed:
fix readme format

jan-kai1 committed:
fix markdown

jan-kai1 committed:
fix markdown table

jan-kai1 committed:
fix markdown table

jan-kai1 committed:
add ui.png

jan-kai1 committed:
fix ui.png

jan-kai1 committed:
changed Chatterbox description

jan-kai1 committed:
changed Chatterbox description

jan-kai1 committed:
Change ui.png

jan-kai1 committed:
Edit readme

jan-kai1 committed:
Userguide and V1 Release

Adds userguide in Readme to provide user a guide in how to use the
various commands for Chatterbox. V1 release also adds an icon, window
resizing and various response changes to better fit Chatterbox
personality.

jan-kai1 committed:
Shorten method lengths

jan-kai1 committed:
Resolve merge conflicts with remote

jan-kai1 committed:
Revert "Change ui.png"

This reverts commit 19a6eacab5f9b02ee2713e7b4a22a5bf7863cd73.

jan-kai1 committed:
Change conditionals for checking empty sets

jan-kai1 committed:
Add new DateTimeFormatters to Parser

jan-kai1 committed:
Merge branch 'master' of https://github.com/jan-kai1/ip

jan-kai1 committed:
Rename ui.png to Ui.png and style fixes

jan-kai1 committed:
Change formatting

jan-kai1 committed:
Add a background for gui

jan-kai1 committed:
Change history storage filename

jan-kai1 committed:
Replace Ui.png

jan-kai1 committed:
Change Background image and Ui.png

jan-kai1 committed:
Fix style issues in fxml

jan-kai1 committed:
Fix readme releases link

jan-kai1 committed:
Add additional input check for Event

jan-kai1 committed:
Add new input check for deadline

jan-kai1 committed:
Revert "Add new input check for deadline"

This reverts commit 5f9efb8acbcce20b3083c66ee5c134a5b203d80c.

jan-kai1 committed:
Add guard against empty input for extractNum

jan-kai1 committed:
Change fxml version in fxml file

jan-kai1 committed:
Fix Readme rendering

jan-kai1 committed:
Update JavaDoc Header Comments for clarity

jan-kai1 committed:
Update JavaDoc Header Comments for clarity

jan-kai1 committed:
Fix and enhance JavaDoc comments across the codebase

jan-kai1 committed:
Fix test cases to handle unordered tags in StorageTest
